src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9972,100 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+/// Return the list of the commands in the specified buffer,
+/// if no buffer is specified, then return list of global(non-buffer) the
+/// commands
+///
+/// @param[in] buf the specified buffer, if NULL then global commands required
+///
+/// @return The array of the commands and the information of each command
+ArrayOf(Dictionary) commands_array(buf_T *buf)
+{
+  Array rv = ARRAY_DICT_INIT;
+  int i = 0, j = 0;
+  char Arg[2] = "";
```

Comment:
this one could also be declared prior to use. Also use lowercase, and `{0,0}` is preferable here (as the second 0 is significant).

Reply:
OK, sure, declaring inside loop is better